server.js

```diff
@@ -45,157 +53,164 @@ function main () {
+                    champMasteries[i].name = championsMap[champId][1];
+                    champMasteries[i].riotName = championsMap[champId][0];
+                    champMasteries[i].urlImage = urlSquare;
+
+                    champId = championsMap[champId];
+                  }
+                  toSend.champMasteries = champMasteries;
+                  return toSend;
+                }
+              }
             }
         });
     });
 }
 
-function getGameInfo(ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    async.waterfall([
-        async.apply(getApiKey, pathToYourApiKey, ApiKey, toSend, championsMap, region, summonerName, version),
-        getVersion,
-        getChampionsJson,
-        getPlayerInfo
-    ], function (err, ApiKey, toSend, championsMap, region, summonerName, version) {
-        callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
-    });
-}
+function getGameInfo(region) {
+  function logError(e) {
+    console.error("error is:" + e);
+    throw (e);
+  }
+  var a = getVersion(region);
+  var b = a.then(function (version) {
+    let req = "http://ddragon.leagueoflegends.com/cdn/" + version + "/data/en_US/champion.json " ;
 
-function getApiKey(file, ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    ApiKey = require(file);
-    callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
-}
+    return new Promise(function(resolve, reject) {
+      request(req, function (error, response, body) {
+        let championsMap = {};
 
-function getVersion (ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    let req = "https://global.api.pvp.net/api/lol/static-data/" + region + "/v1.2/versions?api_key=" + ApiKey;
-    request(req, function (error, response, body) {
         if (!error && response.statusCode == 200) {
-            let str = JSON.parse(body);
-            version = str[0];
-            callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
+          let str = JSON.parse(body);
+          delete str.type;
+          delete str.format;
+          delete str.version;
+          str = str.data;
+          for (let champ in str) {
+            let champTmp = [];
+            champTmp.push(str[champ].id);
+            champTmp.push(str[champ].name);
+            championsMap[str[champ].key] = champTmp;
+          }
+          resolve(championsMap);
         }
         else {
-            if (error && response) {
-                console.log("Problem with the request, error is: " + error + " with status code " + response.statusCode);
-            }
+          console.log("ERROR");
+          if (error && response) {
+            console.log("Problem with the request, error is: " + error + " with status code " + response.statusCode);
+          }
+          return Promise.resolve("Fail");
```

Comment:
It's an error here, you shouldn't resolve the promise, else use something like: `reject(new Error('Bad response from the API')`.